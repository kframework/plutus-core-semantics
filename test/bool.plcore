(program

(module Foo (imported) (exported () ()) (local () ())
  (define result
    (let
      (lam x
        (case x
          (Bool.True  () (con Bool.False))
          (Bool.False () (con Bool.True))))
        negate
        [negate [negate (con Bool.True)]])))

)
