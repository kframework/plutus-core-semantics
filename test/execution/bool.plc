(program

(module Foo (import) (export () ())
  (define result
    (let
      (lam x
        (case x
          (Prelude.True  () (con Prelude.False))
          (Prelude.False () (con Prelude.True))))
        negate
        [negate [negate (con Prelude.True)]])))

)
