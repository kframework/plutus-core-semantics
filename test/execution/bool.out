<T> <k> . </k> <lenv> .Map </lenv> <genv> ( Foo . result ) |-> ( con Bool . True .ValList ) </genv> <mod> Foo </mod> </T>
